XpringKit/XRPTransaction.swift

```diff
@@ -18,6 +19,8 @@ public struct XRPTransaction: Equatable {
   public let sourceTag: UInt32?
   public let transactionSignature: Data
   public let type: XRPTransactionType
+  public let timestamp: UInt64?
+  public let deliveredAmount: String?
```

Comment:
Just curious, when does timestamp get populated? When a transaction gets validated?

Reply:
The answer to this is proving more elusive than I would have thought.  

On [this page](https://xrpl.org/transaction-basics.html#example-executed-transaction-with-metadata), in the example JSON response, we see a `date` field (represented as seconds after the Ripple epoch).  However, I haven't been able to find any discussion of this field in any of the other `xrpl.org` pages. 

According to a similar question I had for Elliot in the context of `ripple-lib`, it looks like transactions *sometimes* have a date, and if they don't, then `ripple-lib` uses the ledger's closing time as the timestamp for the transaction and returns that.  Elliot points out in a [slack archive](https://ripplelabs.slack.com/archives/C02GE9NBA/p1550776640039600?thread_ts=1550775985.037200&cid=C02GE9NBA) that lowercase fields are synthetic fields, whereas upper-case fields are generated directly from the server... `date` is a lower-case field, so it seems likely that transactions don't actually have a date on the XRP ledger, and that the ledger's close time is what's being subbed in for that.  Intuitively, this would make sense to me because a transaction isn't valid until it's included on a closed ledger and in that sense doesn't really exist in a meaningful way yet.

I'm asking Nik Bougalis.

Reply:
Confirmed:  it's the close time of the ledger in which the transaction was included.